### ISSUE TITLE
 Unable to compile Open3D from source on Ubuntu 18.04,undefined reference errors to Assimp

### ISSUE THREAD

**Issue by manhha1402** [intel-isl/Open3D#2170]
**IMPORTANT: Please use the following template to report the bug.**
_____

**Describe the bug**
 I compile Open3D from source. It gives undefined reference errors with Assimp library.

**To Reproduce**
Compile with CMake : `cmake -DGLIBCXX_USE_CXX11_ABI=ON -DPYTHON_EXECUTABLE=/usr/bin/python3 ..`

**Screenshots**
![Screenshot from 2020-08-13 18-00-15](https://user-images.githubusercontent.com/20105284/90158464-4ec7cc00-dd8f-11ea-839c-86f9f03d8482.png)

**Environment (please complete the following information):**

- Operating system: Ubuntu 18.04
- Python version: Python 3.6
- Open3D version: master branch
- Is this remote workstation?:  no
- How did you install Open3D?:  build from source
- Compiler version (if built from source):  gcc 7.5

**Comment by yxlao** [comment]
i tried your config flags and it worked fine for me, @errissa could you take a look as well?

**Comment by errissa** [comment]
I also had no issues with those config flags. @manhha1402 From the build directory would you please share the file assimp/src/ext_assimp-build/CMakeCache.txt

**Comment by manhha1402** [comment]
Hi @errissa ,
Here is the file assimp/src/ext_assimp-build/CMakeCache.txt
[CMakeCache.txt](https://github.com/intel-isl/Open3D/files/5077339/CMakeCache.txt)

**Comment by nachovizzo** [comment]
I'm also facing the exact same issue.

Apparently this was introduced by the new `libassimp` lib dependency in 703260d446. I'm currently trying to figure out what's the issue

**Comment by errissa** [comment]
@manhha1402 Thanks for sharing the file. It was very helpful. I will get back to you with a workaround until we can implement a permanent solution.

**Comment by nachovizzo** [comment]
So, apparently this function is defined in `minzip` that it's within `ziblib` https://github.com/madler/zlib/blob/cacf7f1d4e3d44d871b605da3b647f07d718623f/contrib/minizip/unzip.c#L793.

It looks to be that the `zlib` version that ships with `Open3D` does not include `minzip`, which it is indeed a dependency of the `libassimp` library:
```
../../assimp/lib/libassimp.a(IFCLoader.cpp.o): In function `Assimp::IFCImporter::InternReadFile(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, aiScene*, Assimp::IOSystem*)':
/home/ivizzo/dev/libs/Open3D/build/assimp/src/ext_assimp/code/Importer/IFC/IFCLoader.cpp:186: undefined reference to `unzOpen'
/home/ivizzo/dev/libs/Open3D/build/assimp/src/ext_assimp/code/Importer/IFC/IFCLoader.cpp:203: undefined reference to `unzGoToFirstFile'
/home/ivizzo/dev/libs/Open3D/build/assimp/src/ext_assimp/code/Importer/IFC/IFCLoader.cpp:208: undefined reference to `unzGetCurrentFileInfo'
/home/ivizzo/dev/libs/Open3D/build/assimp/src/ext_assimp/code/Importer/IFC/IFCLoader.cpp:214: undefined reference to `unzOpenCurrentFile'
/home/ivizzo/dev/libs/Open3D/build/assimp/src/ext_assimp/code/Importer/IFC/IFCLoader.cpp:220: undefined reference to `unzReadCurrentFile'
/home/ivizzo/dev/libs/Open3D/build/assimp/src/ext_assimp/code/Importer/IFC/IFCLoader.cpp:233: undefined reference to `unzCloseCurrentFile'
/home/ivizzo/dev/libs/Open3D/build/assimp/src/ext_assimp/code/Importer/IFC/IFCLoader.cpp:247: undefined reference to `unzClose'
```

I manually fixed this issue adding my **system-wide** installation of `libminizip` and it did work indeed.

I've created a branch with a potential solution: I added the `minizip` module to the `zlib` library. Withouth `WIN32` support. Just removing the `iowin32.*` files. And it's indeed working.

What I don't understand is how it's working on the ubuntu CI/CD

**Comment by errissa** [comment]
@nachovizzo If libminizip is not installed on the host system then assimp configures itself to build without it. The base Ubuntu 18.04 does not libminizip installed by default.

**Comment by nachovizzo** [comment]
Ok great, I've created #2184 in the meantime as a workaround. Please let me know if you solve it in another way so I can close the PR

**Comment by yxlao** [comment]
Confirmed, this affects users with `libminizip-dev` installed in the system.
Proposed a fix in #2188. @errissa

**Comment by nachovizzo** [comment]
@yxlao That's a much better fix, thanks! I'll close the other PR
